### DISCUSSION TITLE
Dask Filling Up Available Disk Space

### DISCUSSION THREAD

**Discussion by timhdesilva** [dask/dask#9836] [Q&A]
Hi,

I'm relatively new to Dask, so apologies in advance if this is a dumb or already answered question, but I haven't been able to find anything online to help.

I have a fairly complex set of Dask operations, including several merges of Dask DataFrames on non-index columns, that result in a pandas DataFrame that fits in main memory no problem. My problem is that this set of operations needs to be repeated several times and when I attempt to do that Dask fills up my available disk space in the C: drive (in AppData/Local). My question is whether there is a simple way to clear all the disk space dask is using in between each iteration of the operations that result in a pandas DataFrame? My current solution is to just restart the computer (this is the only way I can clear the disk space) in between several iterations of the operation so I never fill up the disk. This is obviously a pain, so I'm looking for suggestions on how to fix this! 

Thank you!

**Comment by jrbourbeau** [comment]
> My question is whether there is a simple way to clear all the disk space dask is using in between each iteration of the operations that result in a pandas DataFrame?

There's not a simple mechanism for explicitly clearing the disk space. However, I would use `Client.restart()` in between iterations. This resets the entire state of the cluster, so is doing more than just clearing the data stored on workers, but this should handle that too. 

> My problem is that this set of operations needs to be repeated several times and when I attempt to do that Dask fills up my available disk space in the C: drive (in AppData/Local)

I'm curious why things are remaining on disk after your computation is over. Are you holding references to keys that have been spilled to disk?

**Reply by timhdesilva** [reply]
Hi @jrbourbeau,

Thanks for the thoughtful response. Good suggestion with `Client.restart()` - I will give that a try. I'm running this on a machine with minimal resources (due to confidentiality reasons), so I wasn't using `Client()` and was just using the single-machine scheduler by default (was just doing `import dask.dataframe`). 

On your second question, what do you mean by "holding reference to keys that have been spilled over to disk"? Prior to starting the next iteration of my operations, I delete all objects (e.g. Dask and Pandas DataFrames) that were created in the past iteration using `del`. Is there something else I should be doing?

Thanks again!
